core/txpool/legacypool/legacypool.go

```diff
@@ -1733,16 +1733,26 @@ func (pool *LegacyPool) demoteUnexecutables() {
 		}
 		pendingNofundsMeter.Mark(int64(len(drops)))
 
+		// Drop all transactions that were rejected by the miner
+		rejectedDrops := list.txs.Filter(func(tx *types.Transaction) bool {
+			return tx.Rejected()
+		})
+		for _, tx := range rejectedDrops {
+			hash := tx.Hash()
+			log.Trace("Removed rejected transaction by the miner", "hash", hash)
+			pool.all.Remove(hash)
```

Comment:
Technically the trace is not accurate until after `pool.all.Remove`. By no means blocking, as this appears not to have a chance of error.